### PR TITLE
br: add incremental recover/backup.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed, added or deleted? (Required)

I added how-to of the incremental backup/restore to br.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

File: reference/tools/br/br.md [hunk 1]
```diff
@@ -359,6 +359,28 @@ br restore full \
...
+注意增量备份有两个限制：
+- 增量备份需要与前一次全量备份在不同的路径下
+- 增量备份开始时间与 `lastbackupts` 之间不能有 GC
+
+{{< copyable "shell-regular" >}}
+
+```shell
+    LAST_BACKUP_TS=`./br validate decode --field="end-version" -s local:///home/tidb/backupdata`
+    ./br backup full\
+        --pd 172.16.5.xx:2379 \
```
Comment (3pointer) [inline reference/tools/br/br.md:375]:
```suggestion
        --pd ${PDIP}:2379 \
```

File: reference/tools/br/br.md [hunk 2]
```diff
@@ -359,6 +359,28 @@ br restore full \
...
+        -s local:///home/tidb/backupdata/incr \
+        --lastbackupts ${LAST_BACKUP_TS}
+```
+
+以上命令会备份 `[LAST_BACKUP_TS, current PD timestamp)` 之间的增量数据；注意你可以使用 `validate` 指令获取上一次备份的时间戳。
+
+在增量恢复的时候，使用 BR 的方法和普通的恢复并无二样；只不过需要注意恢复增量数据的时候，需要保证备份时指定的 last backup ts 之前备份的数据已经全部恢复到目标集群。
+
```
Comment (3pointer) [inline reference/tools/br/br.md:383]: 可以加一段原理描述，增量备份时候会备份[LAST_BACKUP_TS, current PD timestamp)之间的 DDL，然后恢复时候，会首先执行这些 DDL。

**Comment by TomShawn** [comment]
@YuJuncen Does this PR apply to v3.1?

**Comment by YuJuncen** [comment]
> @YuJuncen Does this PR apply to v3.1?

It would be OK to take this PR to v3.1. @TomShawn

File: reference/tools/br/br.md [hunk 3]
```diff
@@ -359,6 +362,31 @@ br restore full \
...
+    LAST_BACKUP_TS=`./br validate decode --field="end-version" -s local:///home/tidb/backupdata`
+    ./br backup full\
+        --pd ${PDIP}:2379 \
+        -s local:///home/tidb/backupdata/incr \
+        --lastbackupts ${LAST_BACKUP_TS}
+```
+
+以上命令会备份 `[LAST_BACKUP_TS, current PD timestamp)` 之间的增量数据；注意你可以使用 `validate` 指令获取上一次备份的时间戳。
+
+我们备份的增量数据包括 `[LAST_BACKUP_TS, current PD timestamp)` 的新写入，以及这段时间的 DDL。在恢复的时候，我们会先把所有 DDL 恢复，而后才会恢复写入。
```
Comment (TomShawn) [inline reference/tools/br/br.md:386]:
```suggestion
示例备份的增量数据包括 `[LAST_BACKUP_TS, current PD timestamp)` 之间的新写入数据，以及这段时间内的 DDL。在恢复的时候，我们会先把所有 DDL 恢复，而后才会恢复写入数据。
```

File: reference/tools/br/br.md [hunk 4]
```diff
@@ -359,6 +362,31 @@ br restore full \
...
+
+在增量恢复的时候，使用 BR 的方法和普通的恢复并无二样；只不过需要注意恢复增量数据的时候，需要保证备份时指定的 last backup ts 之前备份的数据已经全部恢复到目标集群。
```
Comment (TomShawn) [inline reference/tools/br/br.md:388]: `普通的恢复`就是全量恢复吗？
```suggestion
在增量恢复的时候，使用 BR 的方法和普通的恢复并无差别。需要注意，进行恢复增量数据的时候，需要保证备份时指定的 `last backup ts` 之前备份的数据已经全部恢复到目标集群。
```

Comment (YuJuncen) [reply reference/tools/br/br.md:388]: 是的。

Comment (TomShawn) [reply reference/tools/br/br.md:388]: 请使用更准确的用词。

**Comment by TomShawn** [comment]
LGTM

**Pull request merged**